qutip/solver/heom/bofin_fit.py

```diff
@@ -0,0 +1,1011 @@
+        self.Q = Q
+        self.T = T
+        self.set_spectral_density(w, J)
+
+    def set_spectral_density(self, w, J):
+        """
+        Sets the spectral density to be fitted. It may be provided either as an
+        array of function values or as a python function. For internal reasons,
+        it will then be interpolated or discretized as necessary.
+        """
+
+        if callable(J):
+            self._w = w
+            self._J_array = J(w)
+            self._J_fun = J
+        else:
+            self._w = w
+            self._J_array = J
+            self._J_fun = InterpolatedUnivariateSpline(w, J)
+
+    @staticmethod
+    def _meier_tannor_SD(w, a, b, c):
+        r"""
+        Underdamped spectral density used for fitting in Meier-Tannor form
+        (see Eq. 38 in the BoFiN paper, DOI: 10.1103/PhysRevResearch.5.013181)
+
+        $J(\omega) = \sum_{i=1}^{k} \frac{2 \alpha_{i}^{2} \Gamma_{i} \omega
+        }{\left( \left( \omega + \Omega_{i}\right)^{2} + \Gamma_{i}^{2}
+        \right)+\left( \omega - \Omega_{i}\right)^{2} + \Gamma_{i}^{2}
+        \right)}$
+
+        Parameters
+        ----------
+        w : np.array
+            The frequency of the spectral density
+        a : np.array
+            Array of coupling constant
+        b : np.array
+            Array of cutoff parameters
+        c : np.array
+            Array of resonant frequencies
+        """
+
+        return sum((2 * ai * bi * w
+                    / ((w + ci) ** 2 + bi ** 2)
+                    / ((w - ci) ** 2 + bi ** 2))
+                   for ai, bi, ci in zip(a, b, c))
+
+    def get_fit(
+        self,
+        N=None,
+        Nk=1,
+        final_rmse=5e-6,
+        lower=None,
+        upper=None,
+        sigma=None,
+        guesses=None,
+    ):
+        """
+        Provides a fit to the spectral density with N underdamped oscillator
+        baths. N can be determined automatically based on reducing the
+        normalized root mean squared error below a certain threshold.
+
+        Parameters
+        ----------
+        N : optional, int
+            Number of underdamped oscillators to use.
+            If set to None, it is determined automatically.
+        Nk : optional, int
+            Number of exponential terms used to approximate the bath
+            correlation functions, per bath. Defaults to 5.
+        final_rmse : float
+            Desired normalized root mean squared error.
+        lower : list
+            lower bounds on the parameters for the fit. A list of size 3,
+            each value represents the lower bound for each parameter.
+            The order of the parameters is the same as for the function to be
+            fitted (_meier_tannor_SD). The first term describes the coupling,
+            the second the cutoff frequency,and the last one the central
+            frequency. The lower bounds are considered to be the same for all
+            N modes. for example
```

Comment:
Thank you for the explanation. However, better remove the mention of _meier_tannor_SD, since that method will not appear in the documentation.